### ISSUE TITLE
MMQTTException Error Handling

### ISSUE THREAD

**Issue by DJDevon3** [adafruit/Adafruit_CircuitPython_MiniMQTT#163]
Circuit Python 8.0.5 stable release
```py
try:
        io.loop()
    except (MMQTTException) as e:
        print("MMQTTException: \n", e)
        time.sleep(300)
        continue
```
MMQTTException not behaving as expected.  Does not fail gracefully and continue.  The script stops with an MMQTTException error. Possible I'm doing it wrong.  

Danh suggested rolling it into RuntimeError.  
`I think MMQTTException is just not a subclass of ValueError or RuntimeError. So you want to catch more classes of exceptions than what you have now, or we should make MMQTTException be a subclass of RuntimeError.`

Expected Behavior:
Fail like adafruit_requests library, ignore the error, continue the script after x amount of time, and continue polling the broker server (AdafruitIO)

**Comment by DJDevon3** [comment]
As Danh pointed out I'm missing an import for the error handler. It should be
```py
from adafruit_minimqtt.adafruit_minimqtt import MMQTTException
```
`import adafruit_minimqtt.adafruit_minimqtt as MQTT` at the top of my script [should also include the import ](https://github.com/adafruit/Adafruit_CircuitPython_AdafruitIO/blob/52d17fd0068c1fdd1f0d54635a9d4e795c1af274/adafruit_io/adafruit_io.py#L25)? Seems like a double import for some reason that shouldn't be necessary.

**Comment by brentru** [comment]
@DJDevon3 Agreed that it looks like a double import as adafruit_io.py [already imports MMQTTException](https://github.com/adafruit/Adafruit_CircuitPython_AdafruitIO/blob/52d17fd0068c1fdd1f0d54635a9d4e795c1af274/adafruit_io/adafruit_io.py#L25).

Does the inclusion of `from adafruit_minimqtt.adafruit_minimqtt import MMQTTException` in your `code.py` resolve this error?

**Comment by DJDevon3** [comment]
@brentru this issue was part of [#7907](https://github.com/adafruit/circuitpython/issues/7907) I was beta testing on a specific board.  I'm no longer testing but from what I can remember yes it helped but the error handling still stopped the script even though I was catching the correct one. I can't explain it other than that, it doesn't behave as I would normally expect with a `continue`.  It doesn't actually continue,

**Comment by extrasleepy** [comment]
Hello, 

I'm having a similar MMQTTException error to what's described above.  I was referred to this open issue by the Adafruit Forum. I've been trying multiple "try..except" statments all of which result in a fatal error and a power reset on the microcontroller. 

I recently tried:
except (MMQTTException, MQTTException) as mqttissue: 
to no avail.

Here are my specifics:

Hardware setup:   Adafruit ESP32-S2 Feather with a few I2C sensors
CircuitPython version: the latest version 8.0.5

Library version: circuitpython-bundle-8.x-mpy-20230328

The code: https://gist.github.com/extrasleepy/875dbc9ff177301e50a94c7d4aeec01f

The error message:
![Screen Shot 2023-04-30 at 2 00 50 PM](https://github.com/adafruit/Adafruit_CircuitPython_MiniMQTT/assets/3743846/05904a79-c849-4841-9eaa-28b31de3add6)
![Screen Shot 2023-04-26 at 3 36 28 PM](https://github.com/adafruit/Adafruit_CircuitPython_MiniMQTT/assets/3743846/3ec40036-f9b2-492f-a0f1-641af912f952)



Time duration before the problem appears:
It's unpredictable, but the system rarely runs for longer than 48 hours before getting the error. However, the error can happen in less than 24.
